docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,294 @@
+from dataclasses import dataclass, field
+import uuid
+import itertools
+from typing import (
+    Generator,
+    Iterator,
+    Dict,
+    Sequence,
+    Optional,
+    TYPE_CHECKING,
+    Union,
+    Tuple,
+    List,
+)
+
+import uuid
+import scipy.sparse
+import weaviate
+
+from .... import Document
+from ..base.backend import BaseBackendMixin
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+
+
+@dataclass
+class WeaviateConfig:
+    """This class stores the config variables to initialize
+    connection to the Weaviate server"""
+
+    client: Optional[Union[str, weaviate.Client]] = None
+    n_dim: Optional[int] = None
+    name: Optional[str] = None
+
+
+class BackendMixin(BaseBackendMixin):
+    """Provide necessary functions to enable this storage backend. """
+
+    def _init_storage(
+        self,
+        docs: Optional['DocumentArraySourceType'] = None,
+        config: Optional[WeaviateConfig] = None,
+    ):
+        """Initialize weaviate storage.
+
+        :param docs: the list of documents to initialize to
+        :param config: the config object used to ininitialize connection to weaviate server
+        :raises ValueError: only one of name or docs can be used for initialization,
+            raise an error if both are provided
+        """
+
+        from ... import DocumentArray
+
+        self._schemas = None
+
+        if not config:
+            config = WeaviateConfig()
+
+        self.n_dim = config.n_dim or 1
+
+        import weaviate
+
+        if config.client is None:
+            self._client = weaviate.Client('http://localhost:8080')
+        elif isinstance(config.client, str):
+            self._client = weaviate.Client(config.client)
+        else:
+            self._client = config.client
+
+        if config.name is not None and docs is not None:
+            raise ValueError(
+                'only one of name or docs can be provided for initialization'
+            )
+
+        self._schemas = self._load_or_create_weaviate_schema(config.name)
+        self._offset2ids, self._offset2ids_wid = self._get_offset2ids_meta()
+
+        if docs is None or config.name:
+            return
+
+        elif isinstance(
+            docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
+        ):
+            self.extend(Document(d, copy=True) for d in docs)
```

Comment:
Is it needed to do this here ?
Can't we just do `self.extend(docs)` ?

Reply:
I am not aware of a reason to do this,  thus have changed this to `self.extend(docs)`.